File: integration-tests/src/test/java/org/pastalab/fray/it/EventLogger.java
package org.pastalab.fray.it;

import org.pastalab.fray.core.concurrency.operations.Operation;
import org.pastalab.fray.core.logger.LoggerBase;
import org.pastalab.fray.core.scheduler.Choice;

public class EventLogger implements LoggerBase {
    StringBuilder sb = new StringBuilder();

    @Override
    public void executionStart() {
    }

    @Override
    public void newOperationScheduled(Operation op, Choice choice) {
    }

    @Override
    public void executionDone(boolean bugFound) {

    }

    @Override
    public void applicationEvent(String event) {
        sb.append(event);
    }

    @Override
    public void shutdown() {
    }
}

File: junit/src/main/kotlin/EventLogger.kt
package org.pastalab.fray.junit

import org.pastalab.fray.core.concurrency.operations.Operation
import org.pastalab.fray.core.logger.LoggerBase
import org.pastalab.fray.core.scheduler.Choice

class EventLogger : LoggerBase {
  var sb: StringBuilder = StringBuilder()

  override fun executionStart() {}

  override fun newOperationScheduled(op: Operation, choice: Choice) {}

  override fun executionDone(shouldSave: Boolean) {}

  override fun applicationEvent(event: String) {
    sb.append(event)
  }

  override fun shutdown() {}
}
